### PR TITLE
- reduce number of v1 configurations being built on CI

### PR DESCRIPTION
**Pull request by SSE4**
as conan 2.0 is already [released](https://github.com/conan-io/conan/releases/tag/2.0.0), ConanCenterIndex, according to its [policy](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/README.md) to support only latest conan client releases, should gradually start switching to 2.0 as a main configuration.
therefore, it makes sense to start with reducing an amount of v1 configurations being built, e.g. just to GCC 11 for now, to reduce CI load and build times.
after some grace transition period (e.g. a month), ConanCenterIndex may drop generating new v1 packages altogether.
please note, it doesn't break any existing v1 users, consumers, enterprise customers, etc. existing v1 recipes and packages are going to be never deleted, and they will stay on remote forever, always available for consumption.
going forward, new PRs will be no longer required to pass v1 builds, but only to pass v2 builds.
in other words, if v1 build fails, but v2 build passes, PR will be safely merged by the bot. it will not hurt anyone, as previous recipe revision still exists and available for download.
since now, many people will start to use and rely on v2-only features in their recipes, and v1 clients will be unable to execute that recipe code (because feature doesn't exist in v1 at all, or because there are bugs in v1 that aren't going to be fixed any more).

### CODE REVIEW

**Comment by jcar87** [comment]
Hi @SSE4 ! 

Thanks for opening this PR. We indeed have plans along these lines to build more configurations for Conan 2.0 and eventually wind down the service for Conan 1.X.

However, we have not yet decided on the exact timelines or exact configurations. We will make these decisions in the future based on our monitoring, and they will be announced well in advance so that our users can act accordingly.

For the time being, Conan 1.x is still supported and Conan Center still requires PRs to pass the Conan 1.x validation.

**Comment by SSE4** [comment]
@jcar87 why close PR, if you basically agree and it's aligned with your plans?
we can discuss timelines, exact configuration here based monitoring or whatever.
for contributors, supporting both 1.x and 2.x in parallel is a huge maintenance burden, so usage of 1.x should be decreased going forward, given the fact 2.0 is already released in production (thus, 1.x user base is shrinking rapidly).
